polyglot.js

```diff
@@ -185,13 +193,15 @@ class PolyGlot {
                     message.polyglot_unknown = false;
                 if (!message.polyglot_force && message.polyglot_unknown) {
                     let content = html.find(".message-content")
-                    let new_content = content.text().replace(this.regexp_letters, this.randomRune)
+                    let new_content = this.generateRune(message.data.content,game.settings.get('polyglot','salt') === true ? message.data._id : lang)
```

Comment:
The long `game.settings.get('polyglot',), ` etc... always seems to be passed as first argument to generateRune, so it should probably just be done within that function instead, so here it becomes a single `scrambleString(message.data.content)`

Reply:
To be inside the function the whole message (or journal) should be passed as an argument, shouldn it? The lang variable would need to be determined again (effectively calling the getFlag function twice inside the renderchatMessage scope) when that variable is already determined. Also, there are two kinds of fobjects: messages and journals, wich could end up making the code more complex.

Honestly if there was no option for 'Unique Texts' in the configs only the language would be used as salt and it would become a much simpler `this.generateRune(message.data.content,lang)`. I personally prefer languages being consistent organically (the same text has the same script for the same language) I put that behavior just in case someone didnt.

Please give me your toughts, I am new to java.

Reply:
The lang and Id would also need to be passed on to the function. It would become a function with 3 parameters `scrambleString(message.data.content,id,lang)`. Ok?